### PR TITLE
V2 cypress tests

### PR DESCRIPTION
**Pull request by ohadsahar**

### CODE REVIEW

File: core/components/Header/Header.tsx [hunk 1]
```diff
@@ -12,23 +12,26 @@ const Header = () => {
 
   return (
     <HeaderWrapper data-testid="header-box">
-      <DateWrapper>
-        <DayWrapper>
+      <DateWrapper data-testid="header-date-wrapper">
```
Comment (ssharabiappsforce) [inline core/components/Header/Header.tsx]: Remove not in use

File: core/components/Header/Header.tsx [hunk 2]
```diff
@@ -12,23 +12,26 @@ const Header = () => {
...
+        <DayWrapper data-testid="header-day-wrapper">
```
Comment (ssharabiappsforce) [inline core/components/Header/Header.tsx]: Same

Comment (ohadsahar) [reply core/components/Header/Header.tsx]: Removed

File: core/components/Header/Header.tsx (hunk 1 above)
Comment (ohadsahar) [reply core/components/Header/Header.tsx]: Removed

File: core/components/TasksList/TasksList.tsx [hunk 3]
```diff
@@ -67,21 +69,24 @@ const TasksList = ({
...
-                  TaskStatusType.CREATED,
-                ].includes(listName)}
-                task={task}
-                handleAction={(data: string) => handleAction(data, task)}
-                handleEdit={() => handleEdit(task)}
-              />
+              <div data-testid="task-item">
+                <Task
+                  setCurrentEdit={() => setEditTaskId(task.id)}
+                  showCheck={listName !== TaskStatusType.COMPLETED ?? false}
```
Comment (ssharabiappsforce) [inline core/components/TasksList/TasksList.tsx]: false ?? 'Something' return false 
```suggestion
                  showCheck={listName !== TaskStatusType.COMPLETED}
```

File: cypress/e2e/integration/MainPage.cy.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { DateProps } from "interfaces/date_props.interface";
+
+describe("Renders Main page of app", () => {
+  const lengthOfItems = 5;
+  it("Renders correctly", () => {
+    cy.visit("/");
+  });
+
+  it("Testing date", () => {
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: Need more details about the test

File: core/components/TasksList/TasksList.tsx (hunk 3 above)
Comment (ohadsahar) [reply core/components/TasksList/TasksList.tsx]: Removed, forgot to delete it.
It's was something else before this.

File: cypress/e2e/integration/MainPage.cy.ts (hunk 4 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: This test just getting the current date and checking that details are showing on the screen are the same like the object of the date.

File: cypress/e2e/integration/MainPage.cy.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  });
+
+  it("Testing CRUD on 1 Task", () => {
+    cy.findByTestId("task-input-field").type("Item 1");
+    cy.findByTestId("create-task").click();
+    cy.findByTestId("created-tasks-list").children().should("have.length", 1);
+    cy.findByTestId("task-title").should("have.text", "Item 1");
+    cy.findByTestId("pause-task").click();
+    cy.findByTestId("finish-task").click();
+    cy.findByTestId("remove-task").click();
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: Verify the item removed

File: cypress/e2e/integration/MainPage.cy.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  });
+
+  it(`Testing Local storage and CRUD on ${lengthOfItems} tasks`, () => {
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: I don't see a local storage test. If the action influences the local storage, maybe add a test to check it too.

File: cypress/e2e/integration/MainPage.cy.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    for (let i = 0; i < lengthOfItems; i++) {
+      cy.findByTestId("task-input-field").type(`Item ${i + 1}`);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: add .clear() before the type

File: cypress/e2e/integration/MainPage.cy.ts (hunk 6 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: cy.reload() has been reloading the page again and then it's fetching data from local storage

File: cypress/e2e/integration/MainPage.cy.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    cy.findByTestId("search-tasks").type(`Ohad Sahar`);
+    cy.findByTestId("created-tasks-list").children().should("have.length", 0);
+    cy.findByTestId("do-later-tasks-list").children().should("have.length", 0);
+    cy.findByTestId("finished-tasks-list").children().should("have.length", 0);
+    cy.findByTestId("search-tasks").clear();
+    cy.findByTestId("created-tasks-list")
+      .children()
+      .each(($el, index) => {
+        index === 2
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: if {} else {}

File: cypress/e2e/integration/MainPage.cy.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      });
+    cy.findByTestId("search-tasks").type(`Item 3`);
+    cy.findByTestId("created-tasks-list").children().should("have.length", 0);
+    cy.findByTestId("do-later-tasks-list").children().should("have.length", 1);
+    cy.findByTestId("finished-tasks-list").children().should("have.length", 0);
+    cy.findByTestId("search-tasks").clear();
+  });
+
+  it("Testing Reset Progress", () => {
+    cy.wait(2000);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: Remove wait

Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: Added

File: cypress/e2e/integration/MainPage.cy.ts (hunk 8 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: But it's the same just different syntax, still to change it?

File: cypress/e2e/integration/MainPage.cy.ts (hunk 7 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: Added

File: core/components/TasksList/TasksList.tsx [hunk 10]
```diff
@@ -40,6 +41,7 @@ const TasksList = ({
   listName,
   editTaskId,
   setEditTaskId,
+  "data-testid": dataTestId = "task-list-box",
```
Comment (amitmarko) [inline core/components/TasksList/TasksList.tsx:44]:
```suggestion
  "data-testid" = "task-list-box",
```

File: shared/typography/AppTitle.tsx [hunk 11]
```diff
@@ -7,18 +7,20 @@ export interface AppTitleProps {
...
   role?: string;
+  "data-testid"?: string;
 }
 const AppTitle = ({
   title,
   fontWeight = "100",
   color = "white",
   fontSize = "3vw",
   role = "app-title-role",
+  "data-testid": dataTestid = "app-title-box",
```
Comment (amitmarko) [inline shared/typography/AppTitle.tsx:18]:
```suggestion
  "data-testid" = "app-title-box",
```

Comment (ssharabiappsforce) [reply shared/typography/AppTitle.tsx:18]: Syntax Error

**Review by amitmarko — APPROVED** [review]
LGTM

File: cypress/e2e/integration/MainPage.cy.ts (hunk 4 above)
Comment (ssharabiappsforce) [reply cypress/e2e/integration/MainPage.cy.ts]: But the title of the test needs to be clear to understand what the test does without seeing the code

File: cypress/e2e/integration/MainPage.cy.ts (hunk 6 above)
Comment (ssharabiappsforce) [reply cypress/e2e/integration/MainPage.cy.ts]: Pls pull the local storge data and verify that all render to the dom.

File: cypress/e2e/integration/MainPage.cy.ts (hunk 7 above)
Comment (ssharabiappsforce) [reply cypress/e2e/integration/MainPage.cy.ts]:
```suggestion
      cy.findByTestId("task-input-field").clear().type(`Item ${i + 1}`);
```

File: cypress/e2e/integration/MainPage.cy.ts (hunk 8 above)
Comment (ssharabiappsforce) [reply cypress/e2e/integration/MainPage.cy.ts]: let's have uniformity like `line 51`

File: cypress/e2e/integration/MainPage.cy.ts (hunk 7 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: You will it on the new push

File: cypress/e2e/integration/MainPage.cy.ts (hunk 6 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: II'l create another 'it' for this one.

File: cypress/e2e/integration/MainPage.cy.ts (hunk 5 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: Done

File: cypress/e2e/integration/MainPage.cy.ts (hunk 4 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: Oh, Iil add description.

File: cypress/e2e/integration/MainPage.cy.ts (hunk 8 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts]: Done

File: cypress/e2e/integration/MainPage.cy.ts [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    cy.findByTestId("header-day-name").should(
+      "have.text",
+      convertedDate.currentDay
+    );
+  });
+
+  it("Testing CRUD on 1 Task", () => {
+    cy.findByTestId("task-input-field").type("Item 1");
+    cy.findByTestId("create-task").click();
+    cy.findByTestId("created-tasks-list").children().should("have.length", 1);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: I don't like that the number of tasks is explicit. Maybe calculate and make it dynamic

File: cypress/e2e/integration/MainPage.cy.ts [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    cy.findByTestId("task-title").should("have.text", "Item 1");
+    cy.findByTestId("pause-task").click();
+    cy.findByTestId("finish-task").click();
+    cy.findByTestId("remove-task").click();
+    cy.findByTestId("created-tasks-list").children().should("have.length", 0);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts]: Same

**Review by amitmarko — APPROVED** [review]
LGTM

**Pull request merged**